### PR TITLE
feat(tenant): add label with tenant name for each tenant

### PR DESCRIPTION
**Pull request by oliverbaehler**
Initial efforts for the implementation of:
https://github.com/projectcapsule/capsule-proxy/issues/328

### CODE REVIEW

File: controllers/tenant/metadata.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+const (
+	TenantNameLabel = "kubernetes.io/metadata.name"
+)
+
+// Sets a label on the Tenant object with it's name.
+func (r *Manager) ensureMetadata(ctx context.Context, tnt *capsulev1beta2.Tenant) (err error) {
+	tnt.SetLabels(map[string]string{
+		TenantNameLabel: tnt.Name,
+	})
```
Comment (prometherion) [inline controllers/tenant/metadata.go:20]: We should keep the already existing labels, otherwise, we're gonna overwrite the user ones.

**Review by prometherion — CHANGES_REQUESTED** [review]
I dunno why we would need a new recipe, `lint` when we already have `golint`.

File: controllers/tenant/metadata.go (hunk 1 above)
Comment (oliverbaehler) [reply controllers/tenant/metadata.go:20]: I don't agree, using capsule specific labels seems like we are creating just a technical dept for the future. You could say the same thing here, no?
https://github.com/projectcapsule/capsule/pull/911

I would prefer using a well-known label here as well.

**Review by oliverbaehler — COMMENTED** [review]
Tried to play around with pre-commit. But it's annoying. Removed the target.

File: controllers/tenant/metadata.go (hunk 1 above)
Comment (prometherion) [reply controllers/tenant/metadata.go:20]: I didn't explain properly, here.

Let's say I have a Tenant:
```
NAME    STATE    NAMESPACE QUOTA   NAMESPACE COUNT   NODE SELECTOR   AGE
solar   Active                     0                                 15s
```

I'm labeling it for several reasons:
```
$: kubectl label tnt solar my-label=value
tenant.capsule.clastix.io/solar labeled

$: kubectl get tnt --show-labels
NAME    STATE    NAMESPACE QUOTA   NAMESPACE COUNT   NODE SELECTOR   AGE   LABELS
solar   Active                     0                                 49s   my-label=value
```

With the `tnt.SetLabels` provided by your code we're gonna overwrite the user labels, such as `my-label`.

Comment (oliverbaehler) [reply controllers/tenant/metadata.go:20]: @prometherion Ah right, sorry for the missunderstanding!

**Review by prometherion — APPROVED** [review]

**Pull request merged**